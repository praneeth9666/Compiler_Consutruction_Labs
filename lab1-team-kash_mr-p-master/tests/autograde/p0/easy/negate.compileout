
.globl main
main:
    pushl %ebp        ## save caller’s base pointer
    movl %esp, %ebp   ## set our base pointer
    subl $20, %esp    ## allocate for local vars (replace 20 with the actual size needed)
    pushl %ebx        ## save callee saved registers
    pushl %esi
    pushl %edi
    movl -4(%ebp), %ecx
    negl %ecx
    movl %ecx, -8(%ebp)
    pushl -4(%ebp)
    call print_int_nl
    addl $4, %esp
    pushl -12(%ebp)
    call print_int_nl
    addl $4, %esp

    popl %edi        ## restore callee saved registers
    popl %esi
    popl %ebx
    movl $0, %eax    ## set return value
    movl %ebp, %esp  ## restore esp
    popl %ebp        ## restore ebp (alt. “leave”)
    ret              ## jump execution to call site

