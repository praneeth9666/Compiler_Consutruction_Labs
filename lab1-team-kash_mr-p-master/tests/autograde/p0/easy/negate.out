112
-136332981
